### ISSUE TITLE
PhantomJS2.0, QFontMEtrics::width with default font on OS X 10.10 problem

### ISSUE THREAD

**Issue by rdpanek** [ariya/phantomjs#12918]
Have a good day.

OS: Yosemite 10.10.1
Phantomjs2.0: (a2912c216d06df4d8b51f12ad4082a48c5fc7ba6)

My test is stopped when an error:

```
Assertion failed: (_consumed <= scratch_size), function _hb_coretext_shape, file src/hb-coretext.cc, line 764.
Abort trap: 6
```

and I found a possible cause:
https://bugreports.qt.io/browse/QTBUG-39504

How and when updates QTWebkit after resolved some bugs via QT team?

In phantomjs 1.9.7 is it OK

**Comment by sodabrew** [comment]
@ashkulz Has had recent success moving patches through the Qt code review process. Can you take a look at https://codereview.qt-project.org/#/c/87665/ and advise if we should be testing that patch and trying to revive the patch progress?

**Comment by ashkulz** [comment]
This was merged in a [separate change](https://codereview.qt-project.org/87728) as qtproject/qtbase@abcea1c5f06b2ea0428e08a3a71f08d39c29c2fd -- upgrading to Qt 5.4 or cherry-picking that specific change should be enough to fix the issue.

**Comment by rdpanek** [comment]
@sodabrew @ashkulz Thanks :-) this patch is functional.

**Comment by sodabrew** [comment]
Could we keep this issue open until it is resolved in PhantomJS proper?

**Comment by rdpanek** [comment]
I agree, I'm sorry

**Comment by matAtWork** [comment]
Exact same issue - OSX 10.10.1. Phantom 2.0.0. Did this patch make it into a release/stable build?

**Comment by ashkulz** [comment]
Not sure if @rdpanek submitted a PR for this.

**Comment by ccgagnon** [comment]
Hi, I am running phantomjs by building it form the yosemite branch, but it would be nice to have that fix merged with the 2.0 branch of @ariya repo. When are you planning to do a PR @rdpanek?

**Comment by Vitallium** [comment]
This issue will be fixed with the next Qt upgrade to 5.4.1

**Comment by ashkulz** [comment]
@Vitallium: I would recommend that Qt be upgraded to `5.4.2` instead, will be released within a week's time.

**Comment by lillieatwork** [comment]
Hi @Vitallium & @ashkulz! Thank you for your work on this.

We're running into this same issue and were wondering what the status was on the Qt upgrade?

**Comment by Vitallium** [comment]
@lillieatwork waiting to merge - https://github.com/ariya/phantomjs/pull/13216

**Comment by jakoblind** [comment]
Hi, I seem to have the same issue. What's the status?

**Comment by wflanagan** [comment]
still waiting on this as a fix as well.. FYI.

**Comment by wflanagan** [comment]
Hi all (@Vitallium and @ashkulz in particular).  Seeing that this was merged, I compiled yesterday based on the current master. This error still occurs, at least in that I get the following error: 

Assertion failed: (_consumed <= scratch_size), function _hb_coretext_shape, file src/hb-coretext.cc, line 764

This is different than reported in this ticket, but the same as another ticket that was merged into this one.

**Comment by AngelaO** [comment]
Hi there, I built from the current master last week and still get the following error (only during some calls to phantomjs, not all the time): 

Assertion failed: (_consumed <= scratch_size), function _hb_coretext_shape, file src/hb-coretext.cc, line 764

I'm using OSX 10.10.3

Thanks

**Comment by dbrans** [comment]
+1. I'm running into the same issue.

**Comment by MasazI** [comment]
I have the same issue.

Assertion failed: (_consumed <= scratch_size), function _hb_coretext_shape, file src/hb-coretext.cc, line 764.

**Comment by dbrans** [comment]
Rebuilding phantomjs from source fixed the issue for me.

**Comment by adibbehjat** [comment]
Seconding @dbrans solution. Rebuilding the binary has helped solved the problem. Took a bit of time to finish the job, but it did the job.

**Comment by travsul** [comment]
I have tried rebuilding the source and I'm still receiving the same error afterwards.

**Comment by AngelaO** [comment]
I've also tried rebuilding from source and still get the same error

**Comment by danielpuglisi** [comment]
+1

**Comment by peetucket** [comment]
+1  (with PhantomJS 2.0 on OS X 10.10.5, Ruby 2.2.2)

**Comment by talsafran** [comment]
+1 (with PhantomJS 2.0 on OS X 10.10.1, Ruby 2.2.2)

**Comment by dailystore** [comment]
I'm sorry but what does +1 mean? I'm running into the same issue here.

Assertion failed: (_consumed <= scratch_size), function _hb_coretext_shape, file src/hb-coretext.cc, line 764.
Abort trap: 6

Thanks!

**Comment by mravery** [comment]
@dailystore +1 just means that people are saying that they're "voting" this issue as relevant because they're also encountering the same problem. +1 for me too. Still trying different ways of handling this issue.

**Comment by mravery** [comment]
Here's my fix for it. It comes in two parts. First, in order to build on Yosemite (OX 10.10.*) you need the fix for #12630; there is a patch for that. Second you need this patch: https://github.com/rdpanek/phantomjs/commit/a72df6fbdb2ada7b73aa35aef216b559670486c0

Using `homebrew` I am able to build `phantomjs` because it has the first patch applied (be sure to `brew update` and to use `brew install --build-from-source phantomjs`). But you still need the second patch. You can edit the formula found in `/usr/local/Library/Formula/phantomjs.rb`.

Here are the lines to add:

    patch do
    url "https://raw.githubusercontent.com/Homebrew/patches/480b7142c4e2ae07de6028f672695eb927a34875/phantomjs/yosemite.patch"
      # sha256 "f54bd1592185f031552d3ad5c8809ff27e8f3be4f1c05c81b59bf7dbc4a59de1"
      sha256 "b87a08a30ea4f2716028e5dfe17c77475ae6b9acdfb21c3dac944e593e42e7b7"
    end
    patch do
      url "https://github.com/rdpanek/phantomjs/commit/a72df6fbdb2ada7b73aa35aef216b559670486c0.patch"
      sha256 "72700babeaec47f1a61474495ff736772f5571cb8a913f4b97d3852476a6212e"
    end

I had to change the patch for the Yosemite fix because, for some reason, the patch didn't end with `0x0a` (i.e. carriage return) and that was throwing off the patching system. I renamed the file appropriately so that homebrew could find it.

Then I uninstalled phantomjs and reinstalled it so that homebrew would pick up the changes set in my new formula. For whatever reason, I was not able to build outside of homebrew.

**Comment by khebbie** [comment]
Hello!

 

You have a new message, please read <http://91yuncang.com/gain.php?kq0i>

 

klaus@hebsgaard.dk

**Comment by PierreAmmeloot** [comment]
Hello,

I use "phantomjs-prebuilt@2.1.14", install with NPM (Node.JS). I am on Mac OSX 10.12.2

I have similar issue: 
```
Assertion failed: (_consumed <= scratch_size), function _hb_coretext_shape, file src/hb-coretext.cc, line 764.
Abort trap: 6
```
Latest version don't correct this bug?
